src/core/bridge.ts

```diff
@@ -149,6 +149,22 @@ export abstract class AbstractBridge {
       });
   }
 
+  static normalizeRequest(request: Request): Request {
+    Object.keys(request).forEach((section) => {
+      const callback = AbstractBridge[`normalize${section.charAt(0).toUpperCase() + section.slice(1)}`];
+      if (request[section] && callback) {
+        callback(request, section);
```

Comment:
Should this mutate? What about just returning a new request object of the desired shape?

Reply:
We clone the request (actually I just added a clone for this section, but we clone the request in the other places we set it up) and mutate it already.